lib/modules/manager/npm/post-update/rules.ts

```diff
@@ -20,7 +20,27 @@ export function processHostRules(): HostRulesResult {
     hostType: 'npm',
   });
   logger.debug(`Found ${npmHostRules.length} npm host rule(s)`);
-  for (const hostRule of npmHostRules) {
+  // Include host rules without specific type to mimic the behavior used when determining dependencies with updates.
+  const noTypeHostRules = hostRules
+    .getAll()
+    .filter((rule) => rule.hostType === null || rule.hostType === undefined);
+  logger.debug(
+    `Found ${noTypeHostRules.length} host rule(s) without host type`,
+  );
+  // Drop duplicates for the same matchHost while prefering the more specific rules with hostType npm.
+  const noTypeHostRulesWithoutDuplicates = noTypeHostRules.filter(
+    (rule) => !npmHostRules.some((elem) => elem.matchHost === rule.matchHost),
+  );
+  logger.debug(
```

Comment:
```suggestion
  logger.trace(
```
I think we can drop this log message

Reply:
I'd prefer to log something like this so that this new feature is not "invisible". Perhaps the last message can be lowered

Reply:
So what's the verdict here?